index.html

```diff
@@ -1615,6 +1618,211 @@ <h3>
+        <h2>
+          Usage Example
+        </h2>
+        <p>
+          The following example shows a [=manifest=] for an application which
+          uses the `scope_extensions` member.
+        </p>
+        <pre class="example json" 
+          title="https://example.com/manifest.webmanifest">
+          {
+            "id": "https://example.com/app",
+            "name": "My App",
+            "icons": [{
+              "src": "icon/hd_hi",
+              "sizes": "128x128"
+            }],
+            "start_url": "/app/index.html",
+            "scope": "/app",
+            "display": "standalone",
+            "scope_extensions": [
+              { "type": "origin", "origin": "https://example.co.uk" },
+              { "type": "origin", "origin": "https://help.example.com" }]
+          }
+        </pre>
+        <p>
+          The following shows 2 [=web-app-origin-association=] files which can
+          be downloaded from the `.well-known` path of the [=origins=]  listed 
+          in the `scope_extensions` member.
+        </p>
+        <pre class="example json" 
+        title="https://example.co.uk/.well-known/web-app-origin-association">
+          {
+            "https://example.com/app": {
+              "scope": "/app"   
+            }
+          }
+        </pre>
+        <pre class="example json" 
+        title="https://help.example.com/.well-known/web-app-origin-association">
+          {
+            "https://example.com/app": {
+              "scope": "/"   
+            }
+          }
+        </pre>
+        <p> 
+          The navigation scope of this app consists of URLs that are 
+          [=URL/within scope=] of any of these [=URLs=]:
+          `https://example.com/app`, `https://example.co.uk/app`, and 
+          `https://help.example.com`. 
+        </p>
+      </section>
+      <section>
+      <h2>Processing the scope_extensions member</h2>
+      <p>
+        To <dfn>process the `scope_extensions` member</dfn>, given
+        [=ordered map=] |json:ordered map| and [=ordered map=]
+        |manifest:ordered map|:
+      </p>
+      <ol class="algorithm">
+        <li>Let |processedScopeExtensions:list| be a new [=list=].
+        </li>
+        <li>Set |manifest|["scope_extensions"] to |processedScopeExtensions|.
+        </li>
+        <li>If |json|["scope_extensions"] doesn't [=map/exist=] or is not a 
+          [=list=], return.
+        </li>
+        <li>[=list/For each=] |entry:ordered map| of |json|["scope_extensions"]:
+          <ol>
+            <li>If |entry|["type"] or |entry|["origin"] is undefined,
+              [=iteration/continue=].
+            </li>
+            <li>If |entry|["type"] is not "origin", [=iteration/continue=].
+            </li>
+            <li>If |entry|["origin"] is not a [=string=],
+              [=iteration/continue=].
+            </li>
+            <li>Set |entry|["origin"] to the result of parsing it as an
+              [=origin=].
```

Comment:
There is a spec algorithm for this we should reference. 

The origin of a url is https://url.spec.whatwg.org/#origin

parsing is https://url.spec.whatwg.org/#concept-url-parser


```suggestion
            <li>Set |origin url| to the result of [=URL Parser|parsing=] |entry|["origin"], using the |manifest url| as the base URL]
            </li>
            <li>Set |entry|["origin"] to the [=URL/origin=] of the |origin url|.
```

You'll have to add the manifest url as an argument too. I'm not totally sure offhand about the origin link, but yeah - this should be linked to url parsing & url spec definitions.

Reply:
Wait, these origin input strings aren't supposed to be relative URLs. I'll reference the parsing algo but we don't need a base URL here.

Reply:
SGTM